src/views/Webhook/components/Trigger.vue

```diff
@@ -295,17 +306,34 @@ const validate = () => {
   return Promise.all([validateTopicArr(), checkTopicAllFilled()])
 }
 
+const detectIsMsgEvent = () => {
+  // 1. just msg pub
+  if (fromDataArr.value.every(checkIsTopic) && topicListFromWhere.value.length === 0) {
+    return true
+  }
+  // 2. just other msg event (except msg pub)
+  const isAllMsgEvent = fromDataArr.value.every(
+    (item) => ruleInputEventReg.test(item) && checkIsMsgEvent(item),
+  )
+  if (isAllMsgEvent) {
+    return true
+  }
+  // 3. msg pub and other msg event
+  const isFormDataAllTopicOrMsgEvent = fromDataArr.value.every(
+    (item) => checkIsTopic(item) || checkIsMsgEvent(item),
+  )
+  const isTopicEqualWhere = arraysAreEqual(topicListFromSelect.value, topicListFromWhere.value)
+  return isFormDataAllTopicOrMsgEvent && isTopicEqualWhere
+}
+
 const isSelectedEvent = ref(false)
 const selectedType = computed({
   get() {
     if (!fromDataArr.value.length && isSelectedEvent.value) {
       return TriggerType.Events
     }
-    let msgEventNum = trueTopicList.value.length
-      ? selectedMsgEventList.value.length - 1 + trueTopicList.value.length
-      : selectedMsgEventList.value.length
-    const isAllMsgEvent = msgEventNum === fromDataArr.value.length
-    if (isAllMsgEvent) {
```

Comment:
Is a problem with the logic of judging whether the trigger type is message. I can't understand why I wrote it like this before...

Reply:
This is growth! 🚀